tests/integration/secretsmanager/test_secretsmanager.py

```diff
@@ -335,147 +448,143 @@ def test_rotate_secret_invalid_lambda_arn(self, sm_client):
 
-        rs_get_curr = sm_client.get_secret_value(SecretId=secret_name)
-        assert rs_get_curr["VersionId"] == pv_v0_vid
-        assert rs_get_curr["SecretString"] == secret_string_v0
-        assert rs_get_curr["VersionStages"] == ["AWSCURRENT"]
+        get_secret_value_res_3 = sm_client.get_secret_value(SecretId=secret_name)
+        sm_snapshot.match("get_secret_value_res_3", get_secret_value_res_3)
 
         secret_string_v4: str = "secret_string_v4"
         pv_v4_vid: str = str(uuid.uuid4())
-        pv_v4_res = sm_client.put_secret_value(
+        #
+        put_secret_value_res_4 = sm_client.put_secret_value(
             SecretId=secret_name, SecretString=secret_string_v4, ClientRequestToken=pv_v4_vid
         )
-        assert pv_v4_res["VersionId"] == pv_v4_vid
-        assert pv_v4_res["VersionStages"] == ["AWSCURRENT"]
+        sm_snapshot.match("put_secret_value_res_4", put_secret_value_res_4)
 
-        rs_get_curr = sm_client.get_secret_value(SecretId=secret_name)
-        assert rs_get_curr["VersionId"] == pv_v4_vid
-        assert rs_get_curr["SecretString"] == secret_string_v4
-        assert rs_get_curr["VersionStages"] == ["AWSCURRENT"]
+        get_secret_value_res_4 = sm_client.get_secret_value(SecretId=secret_name)
+        sm_snapshot.match("get_secret_value_res_4", get_secret_value_res_4)
 
-        sm_client.delete_secret(SecretId=secret_name, ForceDeleteWithoutRecovery=True)
+        delete_secret_res_1 = sm_client.delete_secret(
+            SecretId=secret_name, ForceDeleteWithoutRecovery=True
+        )
+        sm_snapshot.match("delete_secret_res_1", delete_secret_res_1)
 
     @pytest.mark.parametrize(
         "secret_name", ["Inv Name", " Inv Name", " Inv*Name? ", " Inv *?!]Name\\-"]
     )
     @pytest.mark.aws_validated
-    def test_invalid_secret_name(self, sm_client, secret_name: str):
-        def check_validation_exception(exc_info: ExceptionInfo):
-            error = exc_info.value.response["Error"]
-            error_code = error["Code"]
-            error_msg = error["Message"]
-            assert error_code == "ValidationException"
-            assert (
-                error_msg
-                == "Invalid name. Must be a valid name containing alphanumeric characters, or any of the following: -/_+=.@!"
-            )
-
+    def test_invalid_secret_name(self, sm_client, sm_snapshot, cleanups, secret_name: str):
+        cleanups.append(
+            lambda: sm_client.delete_secret(SecretId=secret_name, ForceDeleteWithoutRecovery=True)
+        )
         # The secret name can contain ASCII letters, numbers, and the following characters: /_+=.@-
         with pytest.raises(Exception) as validation_exception:
             sm_client.create_secret(Name=secret_name, SecretString="MySecretString")
-        check_validation_exception(validation_exception)
+        sm_snapshot.match("ex_log_1", validation_exception.value)
```

Comment:
```suggestion
        sm_snapshot.match("ex_log_1", validation_exception.value.response)
```

Just saw that I forgot a crucial part of this when suggesting it previously

It should actually be `<exception>.value.response` instead of `<exception>.value`
(Same for the ones below)